src/http-adapter/index.js

```diff
@@ -0,0 +1,203 @@
+  }
+
+  /**
+   * Make a generic request that requires more options then CRUD methods provided.
+   * Does not set default Accept or Content-Type. This must be provided!
+   * @param {string} url - URL where the request is being sent.
+   * @param {Object} options - any options the user wants to pass to the request.
+   * this can contain method, headers, internal, and data
+   * @returns {Promise} Promise relating to request.
+   */
+  async request (
+    url,
+    options = {}
+  ) {
+    options.method = options.method || HTTP_METHOD.GET
+    options.headers = options.headers || {}
+    // set internal to true if this request in intended to go to an interal server to gain the default headers passed into the constructor
+    options.internal = options.internal || false
+
+    // pass in data, even if none is defined to preverve _http method
+    // pass in headers, even if none are defined to override default headers that are set with the _http method
+    return this._http(
+      options.method,
+      url,
+      options.data,
+      options.headers,
+      options.internal
+    )
+  }
+
+  /**
+   * Check if requestion serialization is turned on. If so, the current request will
+   * wait for previous requests to settle. Otherwise, the request will just be executed
+   * @private
+   * @param {Object} -  options passed via destructoring, which can contain url,
+   * method, headers, data, and isInternal properties
+   * @returns {Promise} Promise relating to request resolution.
+   */
+  async _checkSerializeRequests ({
+    url,
+    method,
+    headers,
+    data,
+    isInternal
+  }) {
+    let promise
+    if (this.serializeRequests) {
+      // Wait for all requests to settle (either with success or rejection) before making request
+      const promises = Array.from(this._outstandingRequests).map((promise) =>
+        promise.catch(() => { })
+      )
+
+      promise = Promise.all(promises).then(() =>
+        this._makeRequest({ url, method, headers, data, isInternal })
+      )
+    } else {
+      promise = this._makeRequest({ url, method, headers, data, isInternal });
+    }
+
+    this._outstandingRequests.add(promise)
+    const removeFromOutstandingRequests = () => {
+      this._outstandingRequests.delete(promise)
+    }
+    promise.then(removeFromOutstandingRequests, removeFromOutstandingRequests);
+
+    return await promise
+  }
+
+  /**
+   * Executes an http request
+   * @private
+   * @param {string} method - the type of method being executed. Can be one of the keys of the HTTP_METHOD Object.
+   * @param {String} url - the url of the requwst.
+   * @param {Object} [data] - a data payload represented as a key/value pair.
+   * @param {string} [headers={'Accept': 'application/vnd.api+json', 'Content-Type': 'application/vnd.api+json'}] - headers to be included with the request
+   * @param {string} [isInternal=true] - Whether the request is going to an owned/internal server, in which case header options
```

Comment:
Agreed. I think both are necessary since `isInternal` allows for execution of the request/response interceptors (unless we want to invert that control). Maybe we should, at a minimum, document the `isInternal` option in more detail?

Reply:
After reading over it again, I think `isInternal` is unnecessary and potentially confusing. If I pass a request/response interceptor to the adapter, I would expect it to be used. Perhaps remove the empty function defaults and call them if they're defined?

Reply:
Agreed. I find it to be confusing. Likely will use `requestInterceptors` as an option, but will keep the default functions on the interceptor as it is just easier to handle and likely more type safe